proposals/NNNN-add-customdebugdescription-conformance-to-anykeypath.md

```diff
@@ -0,0 +1,137 @@
+
+## Effect on ABI stability
+
+This proposal will add a new conformance to an existing protocol to an existing type in the stdlib.  
+
+## Effect on API resilience
+
+This change could break existing Swift programs that add their own conformance to `AnyKeyPath`. 
+
+However, it seems unlikely that any Swift program would rely on `debugDescription` in production, when the `description` property is far more appropriate for such a purpose. 
+
+
+## Alternatives considered
+
+### Print fully qualified names or otherwise add more information to the output
+
+ex. `\ModuleName.MyType.myField`, `<KeyPath<MyType, MyFieldType>> \ModuleName.MyType.myField` `(writable) \Theme.backgroundColor`
+
+As this is just for debugging, it seems likely that the information currently being provided would be enough to resolve any ambiguities. If ambiguities arose during a debugging session, in most cases the user could figure out exactly which keypath they were dealing with simply by running `po myKeyPath == \MyType.myField` till they found the right one. 
+
+### Modify KeyPaths to include a string description
+
+This is an obvious solution to this problem, and would likely be very easy to implement, as the compiler already produces `_kvcString`. 
+
+It has the additional advantage of being 100% reliable, to the point where it arguably could be the basis for implementing `description` rather than `debugDescription`. 
+
+However, it would add to the code size of the compiled code, perhaps unacceptably so. Furthermore, it precludes the possiblity of someday printing out the arguments of subscript based keypaths, as
+these can be created dynamically. It would also add overhead to appending keypaths, as the string would also have to be appended. 
+
+I think that most users who might want this _really_ want to use it to build something else, like encodable keypaths. Those features should be provided opt-in on a per-keypath or per-type basis, which will make it much more useful (and in the context of encodable keypaths specifically, eliminate major potential security issues). Such a feature should also include the option to let the user configure this string, so that it can remain backwards compatible with older versions of the program. 
+
+### Make Keypath functions global to prevent the linker from stripping them
+
+This would also potentially make it feasible to change this proposal from implementing `debugDescription` to implementing `description`. 
+
+This would also potentially bloat the binary. It could also be a security issue, as `dlysm` would now be able to find these functions.
+
+I am not very knowlegable about linkers or how typical Swift builds strip symbols, but I think it might be useful to have this as an option in some IDEs that build Swift programs. But that is beyond the scope of this proposal. 
```

Comment:
```suggestion
I am not very knowledgeable about linkers or how typical Swift builds strip symbols, but I think it might be useful to have this as an option in some IDEs that build Swift programs. But that is beyond the scope of this proposal. 
```